pkg/controller/disruption/disruption.go

```diff
@@ -583,10 +605,23 @@ func (dc *DisruptionController) getExpectedScale(pdb *policy.PodDisruptionBudget
 	// 1. Find the controller for each pod.  If any pod has 0 controllers,
 	// that's an error. With ControllerRef, a pod can only have 1 controller.
 	for _, pod := range pods {
+		controllerRef := metav1.GetControllerOf(pod)
+		if controllerRef == nil {
+			err = fmt.Errorf("found no controller ref for pod %q", pod.Name)
```

Comment:
a nil controller didn't log an error previously, right?

Reply:
A pod without a controller was considered an error before this change. We just happened to check all the finder functions before returning the error. This just bails out early if there is no controllerRef.
https://github.com/kubernetes/kubernetes/blob/25b0d2dbf4521bd1e7bae471d85c369eb89f9555/pkg/controller/disruption/disruption.go#L599-L603

Reply:
ah, got it, thanks.